Stripe/STPCard.swift

```diff
@@ -0,0 +1,356 @@
+    return
+      (self.stringToFundingMapping() as NSDictionary).allKeys(
+        for: NSNumber(value: funding.rawValue)
+      ).first as? String
+  }
+
+  // MARK: -
+
+  // MARK: - Equality
+  /// :nodoc:
+  @objc
+  public override func isEqual(_ other: Any?) -> Bool {
+    return isEqual(to: other as? STPCard)
+  }
+
+  /// :nodoc:
+  @objc public override var hash: Int {
+    return stripeID.hash
+  }
+
+  func isEqual(to other: STPCard?) -> Bool {
+    if self === other {
+      return true
+    }
+
+    if other == nil || !(other != nil) {
+      return false
+    }
+
+    return stripeID == other?.stripeID
+  }
+
+  // MARK: - Description
+  /// :nodoc:
+  @objc public override var description: String {
+    let props = [
+      // Object
+      String(format: "%@: %p", NSStringFromClass(STPCard.self), self),
+      // Identifier
+      "stripeID = \(stripeID )",
+      // Basic card details
+      "brand = \(STPCard.string(from: brand))",
+      "last4 = \(last4 )",
+      String(format: "expMonth = %lu", UInt(expMonth)),
+      String(format: "expYear = %lu", UInt(expYear)),
+      "funding = \((STPCard.string(fromFunding: funding)) ?? "unknown")",
+      // Additional card details (alphabetical)
+      "country = \(country ?? "")",
+      "currency = \(currency ?? "")",
+      "dynamicLast4 = \(dynamicLast4 ?? "")",
+      "isApplePayCard = \((isApplePayCard) ? "YES" : "NO")",
+      // Cardholder details
+      "name = \(((name) != nil ? "<redacted>" : nil) ?? "")",
+      "address = \(((address) != nil ? "<redacted>" : nil) ?? "")",
+    ]
+
+    return "<\(props.joined(separator: "; "))>"
+  }
+
+  // MARK: - STPAPIResponseDecodable
+  @objc func stripeObject() -> String {
+    return "card"
+  }
+
+  required init(stripeID: String, last4: String) {
+    self.stripeID = stripeID
+    self.last4 = last4
+    super.init()
+  }
+
+  @objc
+  public class func decodedObject(fromAPIResponse response: [AnyHashable: Any]?) -> Self? {
+    guard let response = response else {
+      return nil
+    }
+    let dict = (response as NSDictionary).stp_dictionaryByRemovingNulls() as NSDictionary
+
+    // required fields
+    guard let stripeId = dict.stp_string(forKey: "id"),
+      let last4 = dict.stp_string(forKey: "last4"),
+      let rawBrand = dict.stp_string(forKey: "brand"),
+      dict.stp_number(forKey: "exp_month") != nil,
+      dict.stp_number(forKey: "exp_year") != nil
+    else {
+      return nil
+    }
+
+    let card = self.init(stripeID: stripeId, last4: last4)
+    card.address = STPAddress()
+
+    card.stripeID = stripeId
+    card.name = dict.stp_string(forKey: "name")
+    card.last4 = last4
+    card.dynamicLast4 = dict.stp_string(forKey: "dynamic_last4")
+    card.brand = self.brand(from: rawBrand)
+    //#pragma clang diagnostic push
+    //#pragma clang diagnostic ignored "-Wdeprecated"
+    // This is only intended to be deprecated publicly.
```

Comment:
- [ ] @yuki-stripe `funding` got un-deprecated in the conversion: Should we just remove it? It seems like it's been deprecated for a while.

Reply:
Actually, it looks like it isn't deprecated in the Objective-C headers.

Reply:
Yeah I'm not sure what this comment is about. I'm going to remove these commented lines